### PR TITLE
Add config setting to allow/deny guests

### LINKED ISSUE #247: Add "no guests" config setting
It would be nice to see a setting within the server configuration which disallows guest users from joining. 

The setting name would be like `authentication_required = true | false`

Ideally, this setting would also be communicated through heartbeat, so that end users would know that they would be able to join the server or not through the server list.

Acceptance Criteria
---
- [ ] Add "no guests" config setting
- [ ] Add equivalent ENV setting
- [ ] Send via heartbeat

### PR DESCRIPTION
**Pull request by SaltySnail**
fix #247

### CODE REVIEW

File: src/TConfig.cpp [hunk 1]
```diff
@@ -51,6 +51,8 @@ static constexpr std::string_view StrAuthKey = "AuthKey";
 static constexpr std::string_view EnvStrAuthKey = "BEAMMP_AUTH_KEY";
 static constexpr std::string_view StrLogChat = "LogChat";
 static constexpr std::string_view EnvStrLogChat = "BEAMMP_LOG_CHAT";
+static constexpr std::string_view StrAllowGuests = "AllowGuests";
+static constexpr std::string_view EnvStrAllowGuests = "BEAMMP_NO_GUESTS";
```
Comment (lionkor) [inline src/TConfig.cpp]: `BEAMMP_ALLOW_GUESTS` please to make it easy to guess the ENV

File: src/TNetwork.cpp [hunk 2]
```diff
@@ -389,6 +389,12 @@ std::shared_ptr<TClient> TNetwork::Authentication(TConnection&& RawConnection) {
             return false;
         });
 
+    beammp_debugf("\t NotAllowedWithReason: {} \t AllowGuests: {} \t IsGuest: {}", NotAllowedWithReason, Application::Settings.AllowGuests, Client->IsGuest());
```
Comment (lionkor) [inline src/TNetwork.cpp]: is this debug print necessary? we could/should remove it or make it a `beammp_tracef` so it only shows up in debug builds with debug mode on

File: src/TNetwork.cpp [hunk 3]
```diff
@@ -389,6 +389,12 @@ std::shared_ptr<TClient> TNetwork::Authentication(TConnection&& RawConnection) {
...
+    if (!NotAllowedWithReason && !Application::Settings.AllowGuests && Client->IsGuest()) { //!NotAllowedWithReason because this message has the lowest priority
+        NotAllowedWithReason = true;
+        Reason = "No guests are allowed on this server! To join, sign up at; forum.beammp.com.";
```
Comment (lionkor) [inline src/TNetwork.cpp]:
```suggestion
        Reason = "No guests are allowed on this server! To join, sign up at: forum.beammp.com.";
```

**Review by lionkor — CHANGES_REQUESTED** [review]
Looks great, good solution there with the low prio deny of the auth, very nice. A few small things to fix ;)

File: src/TConfig.cpp (hunk 1 above)
Comment (SaltySnail) [reply src/TConfig.cpp]: fixed with [6e590ff](https://github.com/BeamMP/BeamMP-Server/pull/335/commits/6e590ff18ad3ea0b09649cdd4652f83003fe70d0)

File: src/TNetwork.cpp (hunk 2 above)
Comment (SaltySnail) [reply src/TNetwork.cpp]: fixed with [e94cfd6](https://github.com/BeamMP/BeamMP-Server/pull/335/commits/e94cfd641d658688335d4789ee42195d440253d3)

**Review by lionkor — APPROVED** [review]
lgtm

**Pull request merged**